### ISSUE TITLE
Move awslambda HttpResponseStream/streamifyResponse to the aws-sdk or its own package?

### ISSUE THREAD

**Issue by mbklein** [aws/aws-lambda-nodejs-runtime-interface-client#74]
The fact that `streamifyResponse` and `HttpResponseStream` are only available embedded in the runtime makes it difficult to write simple, coherent unit tests for streaming lambda handlers. There's been at least [one good attempt](https://github.com/astuyve/lambda-stream) to create a wrapper to support easy testing, but it's not completely interface compatible (lacking, for example, the `HttpResponseStream.from(underlyingStream, prelude)` method) and it seems wasteful to rewrite/repackage code that's already written just to unbundle it from the rest of the NodeJS runtime.

**Comment by sladg** [comment]
Agreed. Having hard time getting any typings and proper tests. At the same time, it seems like it's still under development and we are experiencing changes in behaviour multiple times a week for a past month. Any more details, examples, etc. would be appreciated.
